Resources/public/js/views/serverside/ez-contenttypeserversideview.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-contenttypeserversideview', function (Y) {
+    "use strict";
+    /**
+     * Provides the content type server side view
+     *
+     * @module ez-contenttypeserversideview
+     */
+    Y.namespace('eZ');
+
+    /**
+     * The content type server side view.
+     *
+     * @namespace eZ
+     * @class ContentTypeServerSideView
+     * @constructor
+     * @extends eZ.ServerSideView
+     */
+    Y.eZ.ContentTypeServerSideView = Y.Base.create('contentTypeServerSideView', Y.eZ.ServerSideView, [], {
+        events: {
+            '.ez-relation-pick-root-button': {
+                'tap': '_pickRoot'
+            },
+        },
+
+        /**
+         * tap event handler on the root item selection buttons. It launches the
+         * universal discovery widget so that the user can pick a content.
+         *
+         * @method _pickRoot
+         * @protected
+         * @param {EventFacade} e
+         */
+        _pickRoot: function (e) {
+            var button = e.target,
+                unsetLoading = Y.bind(this._uiUnsetPickRootLoading, this, button);
+
+            e.preventDefault();
+            this._uiSetPickRootLoading(button);
+            this.fire('contentDiscover', {
+                config: {
+                    title: button.getAttribute('data-universaldiscovery-title'),
+                    cancelDiscoverHandler: unsetLoading,
+                    contentDiscoveredHandler: Y.bind(this._setRoot, this, button),
+                    multiple: false,
+                    data: {
+                        afterUpdateCallback: unsetLoading,
+                    },
+                },
+            });
+        },
+
+        /**
+         * Puts picked location into field value
+         *
+         * @method _setRoot
+         * @protected
+         * @param {Y.Node} button
+         * @param {EventFacade} e
+         */
+        _setRoot: function (button, e) {
+            var selectionRoot = button.ancestor('.selection-root');
+
+            selectionRoot.one('input').setAttribute('value', e.selection.location.get('id'));
+            this._uiUnsetPickRootLoading(button);
+        },
+
+        /**
+         * Changes the state of the provided root selection button to be *loading* and
+         * to be disabled
+         *
+         * @method _uiSetPickRootLoading
+         * @protected
+         * @param {Y.Node} button
+         */
+        _uiSetPickRootLoading: function (button) {
```

Comment:
you don't really need that in this case, because once the user has chosen a location in the UDW, we are done, there's no asynchronous process after.